### PR TITLE
Engine: add basic proptest for titan

### PR DESCRIPTION
**Pull request by tonyxuqqi**
### What is changed and how it works?

Issue Number: Ref #16259

What's Changed:

```commit-message
Add property test for titan's basic operations: get/put/delete/scan/delete_range
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by v01dstar — APPROVED** [review]
LGTM

File: components/engine_rocks/Cargo.toml [hunk 1]
```diff
@@ -65,3 +65,4 @@ features = ["encryption"]
 [dev-dependencies]
 rand = "0.8"
 toml = "0.5"
+proptest = "1.0.0"
```
[the remark this answers is not in the document]
Comment (tonyxuqqi) [reply components/engine_rocks/Cargo.toml:68]: No particular reason. We should use latest version.

[the remark this answers is not in the document]
Comment (ekexium) [reply components/engine_rocks/Cargo.toml:68]: Okay I was confused by the rules

**Pull request merged**